cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -15,15 +15,29 @@
 // specific language governing permissions and limitations
 // under the License.
 
+#include <cassert>
+#include <cmath>
+#include <memory>
+#include <type_traits>
+#include <utility>
+
 #include "arrow/compute/api_aggregate.h"
 #include "arrow/compute/kernels/aggregate_basic_internal.h"
 #include "arrow/compute/kernels/aggregate_internal.h"
+#include "arrow/compute/kernels/codegen_internal.h"
 #include "arrow/compute/kernels/common_internal.h"
 #include "arrow/compute/kernels/util_internal.h"
+#include "arrow/type.h"
+#include "arrow/type_traits.h"
+#include "arrow/util/align_util.h"
+#include "arrow/util/bit_block_counter.h"
 #include "arrow/util/cpu_info.h"
+#include "arrow/util/decimal.h"
 #include "arrow/util/hashing.h"
 
-#include <memory>
+// Include templated definitions for aggregate kernels that must compiled here
+// with the SIMD level configured for this compilation unit in the build.
+#include "arrow/compute/kernels/aggregate_basic-inl.cc"  // NOLINT
```

Comment:
What does "inl" mean? Is it for "inline"? If so, can we rename it to `aggregate_basic_inline.cc`?

Reply:
Yes it is short for "inline".

I think "-inl" (or similarly ".inl") is more commonly seen in other cpp projects than "_inline".

Some examples:
https://github.com/google/googletest/blob/main/googletest/src/gtest-internal-inl.h
https://github.com/facebook/folly/blob/main/folly/futures/Future-inl.h
https://github.com/boostorg/thread/blob/master/test/util.inl

Reply:
Perhaps, OTOH "inline" is transparent while "inl" induces a bit of head-scratching (cue "strcat").

Reply:
@pitrou it's a convention as pointed out by @zanmato1984 above. The weird name with `-`  to communicate this is not a regular `.h` or `.cc` file.